ci/check_d_macro_calls.py

```diff
@@ -0,0 +1,404 @@
+                if count_part[0] == '*':
+                    count = count_part[1:].strip()
+                    line.set_alloc(line.alloc_var, count)
+
+        # Check for braces which can be removed.
+        if line.close_brace and lines[idx-2].conditional_brace:
+            line.drop_line()
+            lines[idx-2].remove_brace()
+
+        # Check for assignment of NULL after free.
+        elif line.free_var and \
+             lines[idx+1].is_assign_null(line.free_var):
+            lines[idx+1].drop_line()
+
+        # Check for conditional free.
+        elif line.free_var and \
+             lines[idx-1].is_cond_on_var(line.free_var):
+            if lines[idx-1].conditional_brace:
+                if lines[idx+1].close_brace:
+                    line.shift_left()
+                    lines[idx-1].drop_line()
+                    lines[idx+1].drop_line()
+            else:
+                line.shift_left()
+                lines[idx-1].drop_line()
+
+        # Check for conditional free with other tests.
+        elif line.free_var and \
+             lines[idx-1].is_cond_part_on_var(line.free_var):
+            if lines[idx-1].conditional_brace:
+                if lines[idx+1].close_brace:
+                    lines[idx-1].remove_brace()
+                    lines[idx+1].drop_line()
+            lines[idx-1].shorten_cond(line.free_var)
+
+        if 'test' in line.filename:
+            continue
+        if lines[idx-1].is_todo():
+            continue
+
+        continue
+
+        # These checks need some more work yet, and will require reworking
+        # of some code before they can be used.
+        if line.alloc_var:
+            next_line = idx + 1
+            if not line.text.endswith(';'):
+                continue
+
+            if next_line > len(lines):
+                pass
+            elif lines[next_line].conditional:
+                if not lines[next_line].is_cond_on_null_var(line.alloc_var):
+                    line.mark('Revolve suspect code block')
+            elif lines[next_line].text == '' and len(lines) < next_line and \
+                 lines[next_line+1].conditional:
+                lines[next_line].drop_line()
+            else:
+                line.mark('Add check for alloc fail')
+    show_patch(lines)
+
+CODE_MATCH = ['D_FREE', 'D_ALLOC']
+
+def main():
+    """Check the source tree for D_FREE() usage.
+
+    Check all D_FREE uses, and output a patch to remove any conditional calls
+    to improve logging.
+    """
+
+    args = ['grep', '-n', '-B1', '-A1']
+
+    for match in CODE_MATCH:
+        args.extend(['-e', match])
+
+    if len(sys.argv) == 2:
+        # If a file is passed on the command line then just check that.
+        cmd = args
+        cmd.extend(['-H', sys.argv[1]])
+    else:
+        # Or use git to check the 'src' subdir.
+        cmd = ['git']
+        cmd.extend(args)
+        cmd.append('src')
+
+    rc = subprocess.run(cmd, stdout=subprocess.PIPE)
```

Comment:
(pylint-subprocess-run-check) Using subprocess.run without explicitly set `check` is not recommended.